src/components/CodeEditor/PythonEditor.js

```diff
@@ -0,0 +1,154 @@
+import React, { useState, useEffect } from "react";
+
+import { usePython } from "./react-py";
+
+import CodeMirrorEditor from "./CodeMirror";
+
+import ArrowRightIcon from "@mui/icons-material/ArrowRight";
+import RestartAltIcon from "@mui/icons-material/RestartAlt";
+import CableIcon from "@mui/icons-material/Cable";
+
+import { Box, Button, Grid, Typography } from "@mui/material";
+
+const PythonEditor = ({
+  initialCodeEditorValue,
+  disableEditing,
+  disableRun,
+}) => {
+  let { runPython, stdout, stderr, isLoading, isRunning } = usePython();
+  const [pythonEditorCode, setPythonEditorCode] = useState(
+    initialCodeEditorValue
+  );
+  const [codeRan, setCodeRan] = useState();
+  let [codeExecuted, setCodeExecuted] = useState(stdout, stderr);
```

Comment:
Why are these state variables necessary? Why can't we just use `isLoading` and `isRunning` since every time the values of these variables change, the component should be re-rendered.